code/modules/surgery/asthmatic_bypass.dm

```diff
@@ -0,0 +1,96 @@
+/datum/surgery/asthmatic_bypass
+	name = "Asthmatic Bypass"
+	surgery_flags = SURGERY_REQUIRE_RESTING | SURGERY_REQUIRE_LIMB
+	requires_bodypart_type = NONE
+	organ_to_manipulate = ORGAN_SLOT_LUNGS
+	possible_locs = list(BODY_ZONE_CHEST)
+	steps = list(
+		/datum/surgery_step/incise,
+		/datum/surgery_step/retract_skin,
+		/datum/surgery_step/clamp_bleeders,
+		/datum/surgery_step/incise,
+		/datum/surgery_step/expand_windpipe,
+		/datum/surgery_step/close,
+	)
+
+/datum/surgery/asthmatic_bypass/can_start(mob/user, mob/living/patient)
+	. = ..()
+
+	if (!.)
+		return
+
+	return (patient.has_quirk(/datum/quirk/item_quirk/asthma))
```

Comment:
still using has_quirk as i have significantly decreased the amount of places you check for the asthma quirk